src/mlpack/core/tree/rectangle_tree/hilbert_r_tree_split.hpp

```diff
@@ -1 +1,86 @@
+/**
+ * @file hilbert_r_tree_split.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * Defintion of the HilbertRTreeSplit class, a class that splits the nodes of an R
+ * tree, starting at a leaf node and moving upwards if necessary.
+ */
+#ifndef MLPACK_CORE_TREE_RECTANGLE_TREE_HILBERT_R_TREE_SPLIT_HPP
+#define MLPACK_CORE_TREE_RECTANGLE_TREE_HILBERT_R_TREE_SPLIT_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace tree /** Trees and tree-building procedures. */ {
+
+/**
+ * The order of the splitting policy. The Hilbert R tree splits a node
+ * on overflow, turnung splitOrder node to (splitOrder+1) nodes.
+ */
+constexpr int splitOrder = 2;
```

Comment:
I'll make it a template parameter. I didn't test the values those are not equal to 2. Maybe some other values will be better.